src/include/uapi/ipc/topology.h

```diff
@@ -162,6 +163,18 @@ struct sof_ipc_comp_volume {
 	uint32_t initial_ramp;	/**< ramp space in ms */
 } __attribute__((packed));
 
+/* generic selector component */
+struct sof_ipc_comp_selector {
```

Comment:
This should all use the mux ipc config and source/sink/selected channels should be set at runtime via kcontrol. @ranj063 is integrating this part too.